metaflow/plugins/resources_decorator.py

```diff
@@ -77,5 +93,8 @@ def myStep(self):
         "gpu_vendor": None,
         "memory_limit": None,
         "local_storage": None,
-        "local_storage_limit": None
+        "local_storage_limit": None,
+        "volume": None,
+        "volume_mode": "ReadWriteOnce",
+        "volume_dir": "/opt/metaflow_volume"
```

Comment:
I do wonder if instead of having many `volume_*` parameters we instead make `volume` accept `Union[str, Volume]` where `Volume` is a `dict` or `object` with `mode`, `dir` attributes then it can be easily extended to many volumes and to additional volume attributes.

Reply:
Thanks for bringing this up.

`@resources(volume="11G")`
and in the future, when needed we can add volumes:
`@resources(volumes=[("11G", "ReadWriteOnce", "/opt/metaflow_volume")])`

When we need multiple volumes, we introduce `@resources(volumes=)` while allowing `@resources(volume=)` to make the 80% case easy.

Reply:
Yeah a tuple would work but is less descriptive in my opinion but that can be a future problem :)